### ISSUE TITLE
gcc -broken

### ISSUE THREAD

**Issue by compyl118** [HesselM/rpicross_notes#13]
I copied the issue here for visibility, sorry if it is seen as pollution, for original post clic #11 at the bottom
---------------
Hello, I have the same issue but I do not understand how you solved yours.
Could you give details please?

_Originally posted by @compyl118 in https://github.com/HesselM/rpicross_notes/issues/11#issuecomment-485265255_

**Comment by HesselM** [comment]
There is not much information to go on, but I guess you get stuck at this step: https://github.com/HesselM/rpicross_notes/blob/master/04-xc-setup.md#test-setup ?

Did you deviate much from the described steps?

And what is the cmake output after:
```
cmake -D CMAKE_TOOLCHAIN_FILE=/home/pi/rpicross_notes/rpi-generic-toolchain.cmake ~/rpicross_notes/hello/pi
```

It could be the case that you use a newer gcc-compiler, that cmake is not configured properly or something else..

**Comment by compyl118** [comment]
Hello,

Sorry, I didn't want to elaborate on my issue before I knew I would have a reply. 

Yes I do get stuck at the step you mentionned. Here is the output I get from the cmake command:
![Capture](https://user-images.githubusercontent.com/48632921/56651805-298ae800-668a-11e9-8947-fb752f3fcb6c.PNG)
As you can see it is very similar to what #11 is about butI have some struggle understanding what is happening and what the cmake output means. 

No, I did not deviate from the previous steps. Not knowingly at least. Except for the following:
-I used version 18.04 of the ubuntu live server instead of 16.04 (just realized that btw).

Thank you for considering my issue :)

**Comment by compyl118** [comment]
oups closed by mistake :/

**Comment by HesselM** [comment]
What are the outputs of the following commands:
```
file /usr/bin/rpizero-gcc
```
and
```
/usr/bin/rpizero-gcc -v
```

Edit: could you also test:
```
which make
```
and
```
make -v
```

It looks like the wrong `make` is used (located in `rootfs`, or in other words, the ARM-`make` used on the rpi platform instead of usr/bin/make)

**Comment by compyl118** [comment]
Here is the result for the "file /usr/bin/rpizero-gcc" command:
![Capture](https://user-images.githubusercontent.com/48632921/56657682-f7cd4d80-6698-11e9-80ac-51e8f69ca722.PNG)

For the "/usr/bin/rpizero-gcc -v" command: 
![Capture](https://user-images.githubusercontent.com/48632921/56657847-509ce600-6699-11e9-96b6-50dcd91622ab.PNG)

for the "wich make" command:
![Capture](https://user-images.githubusercontent.com/48632921/56657876-64e0e300-6699-11e9-99ab-c6368445b333.PNG)

For the "make -v" command:
![Capture](https://user-images.githubusercontent.com/48632921/56657980-affaf600-6699-11e9-8d2a-32bfe725c6b2.PNG)

**Comment by HesselM** [comment]
What is the output of CMakeOutput.log and CMakeError.log ? 

(btw, it might be better to copy-paste the terminal output instead of images. It improved readability, but also the possibility of others to finding the issue by googling on the error. You can use triple-quotes to insert code in your comment: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code)

**Comment by compyl118** [comment]
Hello,

sorry for such a delayed response but I was quite busy on other subjects lately :/ 
Here are the logs you asked for (it took me quite a while to figure out how to switch from screenshot to text but I'm glad you made me do it ;) ):
CMakeError.log
```Determining if the C compiler works failed with the following output:
Change Dir: /home/pi/CMakeFiles/CMakeTmp

Run Build Command:"/home/pi/rpi/rootfs/usr/bin/make" "cmTC_579ab/fast"
/home/pi/rpi/rootfs/usr/bin/make: 1: /home/pi/rpi/rootfs/usr/bin/make: Syntax error: word unexpected (expecting ")")


Determining if the C compiler works failed with the following output:
Change Dir: /home/pi/CMakeFiles/CMakeTmp

Run Build Command:"/home/pi/rpi/rootfs/usr/bin/make" "cmTC_43a0d/fast"
/home/pi/rpi/rootfs/usr/bin/make: 1: /home/pi/rpi/rootfs/usr/bin/make: Syntax error: word unexpected (expecting ")")
```

CMakeOutput.log
```The target system is: Linux - 1 - arm
The host system is: Linux - 4.15.0-47-generic - x86_64
Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
Compiler: /usr/bin/rpizero-gcc 
Build flags: ;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include
Id flags:  

The output was:
0


Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"

The C compiler identification is GNU, found in "/home/pi/CMakeFiles/3.10.2/CompilerIdC/a.out"

Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
Compiler: /usr/bin/rpizero-g++ 
Build flags: ;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include
Id flags:  

The output was:
0


Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "a.out"

The CXX compiler identification is GNU, found in "/home/pi/CMakeFiles/3.10.2/CompilerIdCXX/a.out"

Compiling the C compiler identification source file "CMakeCCompilerId.c" succeeded.
Compiler: /usr/bin/rpizero-gcc 
Build flags: ;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include
Id flags:  

The output was:
0


Compilation of the C compiler identification source "CMakeCCompilerId.c" produced "a.out"

The C compiler identification is GNU, found in "/home/pi/CMakeFiles/3.10.2/CompilerIdC/a.out"

Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
Compiler: /usr/bin/rpizero-g++ 
Build flags: ;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include;;-isystem;/home/pi/rpi/rootfs/usr/include/arm-linux-gnueabihf;-isystem;/home/pi/rpi/rootfs/usr/include;-isystem;/home/pi/rpi/rootfs/usr/local/include
Id flags:  

The output was:
0


Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "a.out"

The CXX compiler identification is GNU, found in "/home/pi/CMakeFiles/3.10.2/CompilerIdCXX/a.out"

```

**Comment by compyl118** [comment]
up ?

**Comment by HesselM** [comment]
Ok, it definitely is the wrong make bin: your setup selects the arm-version of “make” (the one in rootfs), while it should use the version for your system (in usr/bin, as your previous reply showed) - your OS is not able to process the contents of the arm-binary, and hence produces the error that it cannot build.

It could be an issue of a changing the order of entries in $PATH or configuring cmake differently. Currently i do not have a solution on the top of my mind (nor in the position of having a test system running to figure out the proper setting).

Did you find a solution in the meantime?

**Comment by HesselM** [comment]
solved by #14
